PocketKit/Sources/SharedPocketKit/Tags/RecentTagsProvider.swift

```diff
@@ -0,0 +1,48 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import Foundation
+
+public class RecentTagsProvider {
+    private let userDefaults: UserDefaults
+    private let key: UserDefaults.Key
+
+    public init(userDefaults: UserDefaults, key: UserDefaults.Key) {
+        self.userDefaults = userDefaults
+        self.key = key
+    }
+
+    /// Handles the list of recent tags from userDefaults
+    public var recentTags: [String] {
```

Comment:
Alternate take:
If we worked with an Ordered Set instead of an Array we could avoid checking for dupes and the like.
Might be more trouble than it's worth, but still a fun thought experiment.

Reply:
thanks! I will think about this for the next PR if that's okay!